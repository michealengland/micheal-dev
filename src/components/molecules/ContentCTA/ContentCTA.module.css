.cta {
  align-items: center;
  background-size: cover;
  box-sizing: border-box;
  color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  min-height: 430px;
  padding: 1.2em;
  position: relative;
  text-align: center;

  @media screen and (min-width: 48em) {
    padding: 1.2em 4rem;
  }

  @media screen and (min-width: 80em) {
    padding: 1.2em 10rem;
  }

  &::before {
    background-color: rgba(2, 0, 65, 0.65);
    bottom: 0;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;
  }

  hr {
    border: none;
    margin: 1.2em 0;
    height: 3px;
    background: #74edf1;
    border-radius: 20px;
    max-width: 140px;
    width: 100%;
  }

  p {
    font-size: 1em;
    line-height: 1.6em;
    margin: 0 0 0.5em 0;
    padding: 0 0 1.2em 0;
  }

  & .inner {
    background-color: #170387;
    margin: 0 auto;
    max-width: 40em;
    padding: 3em 1.2em;
    position: relative;
    text-align: left;
    width: 100%;
    z-index: 1;

    @media screen and (min-width: 60em) {
      padding: 2em;
    }
  }

  & .title {
    @apply p-0;
  }

  & .content {
    @apply p-0;
  }
}
