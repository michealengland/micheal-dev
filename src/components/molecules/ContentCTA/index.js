import React from 'react'
import Heading from '@components/atoms/Heading'
import PropTypes from 'prop-types'
import styles from './ContentCTA.module.css'
import createMarkup from '@utilities/createMarkup'

export default function ContentCTA({content, title, titleTag}) {
  return (
    <section className={styles.cta}>
      <div className={styles.inner}>
        {title && <Heading className={styles.title} tag={titleTag}>{title}</Heading>}
        {content &&
          <>
            <hr />
            <div className={styles.content} dangerouslySetInnerHTML={createMarkup(content)} />
          </>
        }
      </div>
    </section>
  )
}

ContentCTA.propTypes = {
  content: PropTypes.string,
  label: PropTypes.string,
  title: PropTypes.string,
  titleTag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  url: PropTypes.string,
}

ContentCTA.defaultProps = {
  titleTag: 'h2'
}
