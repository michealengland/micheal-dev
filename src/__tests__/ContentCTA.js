import React, {Fragment} from 'react'
import renderer from 'react-test-renderer'
import ContentCTA from '@components/molecules/ContentCTA'

describe('ContentCTA', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Fragment>
          <ContentCTA
            content="Content..."
            label="Button Label"
            title="Content CTA"
            url="https://codepen.io/mengland/pens/"
          />
          <ContentCTA
            content="Content..."
            img="./images/ash-goldsbrough-1092046-unsplash.jpg"
            label="Button Label"
            title="Content CTA with Image"
            url="https://twitter.com/mikelikethebike"
          />
        </Fragment>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
