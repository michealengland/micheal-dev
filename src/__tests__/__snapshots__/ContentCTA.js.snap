// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCTA renders correctly 1`] = `
Array [
  <section
    className="cta"
  >
    <div
      className="inner"
    >
      <h2
        className="title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Content CTA",
          }
        }
      />
      <div
        className="content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Content...",
          }
        }
      />
      <a
        className="button button"
        href="https://codepen.io/mengland/pens/"
      >
        Button Label
      </a>
    </div>
  </section>,
  <section
    className="cta hasImg"
    style={
      Object {
        "backgroundImage": "url(./images/ash-goldsbrough-1092046-unsplash.jpg)",
      }
    }
  >
    <div
      className="inner"
    >
      <h2
        className="title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Content CTA with Image",
          }
        }
      />
      <hr />
      <div
        className="content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Content...",
          }
        }
      />
      <a
        className="button button"
        href="https://twitter.com/mikelikethebike"
      >
        Button Label
      </a>
    </div>
  </section>,
]
`;
